### DISCUSSION TITLE
Ensuring place_agent() only places agent once (to avoid warnings)

### DISCUSSION THREAD

**Discussion by maplesync** [mesa/mesa#2380] [Q&A]
Hi there, 
I am very new to Mesa and am excited about it! However, I am stuck on [lesson 7](https://github.com/SFIComplexityExplorer/Mesa-ABM-Tutorial/blob/main/Session_7_Initialize_Traders.ipynb) of the SFI Complexity Explorer Tutorial on Mesa. I am given the following warning when I run the code to place agents:

`UserWarning: Agent 4088 is being placed with
place_agent() despite already having the position (49, 17). In most
cases, you'd want to clear the current position with remove_agent()
before placing the agent again.`

I have also tested this with the lesson's GitHub code in CoLab and it produces the same result. I see that this warning was added relatively recently, but I am wondering what code I need to use at this stage so that the warning doesn't come up. I want to learn to place only one agent rather than the same agent multiple times (I presume).

Thanks for your help as I am stumped. And thanks for working on this fantastic project. I can't wait to sink my teeth in deeper.

**Comment by EwoutH** [comment]
Hi, thanks for reaching out!

If an Agent is already placed, the convention is to use `move_agent()` instead of `place_agent()`.

**Reply by PestyVesty** [reply]
I had similar warning. I found I had manually set the position for an agent when initially creating the agent, and then placed the agent in the environment. (Essentially placing the agent twice). If my understanding is correct, every agent has a 'pos' variable, which is initially 'None'. After placing or moving an agent, 'pos' will get updated accordingly.
